src/FastConv/FastConv.cpp

```diff
@@ -0,0 +1,130 @@
+
+#include "FastConv.h"
+
+CFastConv::CFastConv( void )
+{
+}
+
+CFastConv::~CFastConv( void )
+{
+    reset();
+}
+
+Error_t CFastConv::init(float *pfImpulseResponse, int iLengthOfIr, int iBlockLength)
+{
+
+    m_ImpulseResponse = pfImpulseResponse;
+    m_blockLength = iBlockLength;
+    m_IRlen = iLengthOfIr;
+    m_pcRingBuff = new CRingBuffer<float>(iLengthOfIr);
+    CFft::createInstance(fft);
+    fft->initInstance(iBlockLength);
+
+    return Error_t::kNoError;
+}
+
+Error_t CFastConv::reset()
+{
+    m_ImpulseResponse = 0;
+//    delete [] m_pcRingBuff;
+    m_ImpulseResponse = 0;
+    m_pcRingBuff = 0;
+
+    return Error_t::kNoError;
+}
+
+Error_t CFastConv::process (float* pfOutputBuffer, const float *pfInputBuffer, int iLengthOfBuffers )
+{
+
+    std::memset(pfOutputBuffer, 0, iLengthOfBuffers);
+    for (int i = 0; i < iLengthOfBuffers; i ++)
+    {
+        m_pcRingBuff->setReadIdx(m_pcRingBuff->getWriteIdx() + 1);
+        m_pcRingBuff->putPostInc(pfInputBuffer[i]);
+
+        for (int j = m_IRlen - 1; j >= 0; j--)
+        {
+            pfOutputBuffer[i] += m_ImpulseResponse[j] * m_pcRingBuff->getPostInc();
+        }
+
+    }
+    return Error_t::kNoError;
+}
+
+Error_t CFastConv::flushBuffer(float* pfOutputBuffer)
+{
+
```

Comment:
Flush buffer for time domain is not implemented, your output of the process is only going to be ilengthofbuffers long, you still need to get the tail which will be impulseresponsesize-1 block length, which will flush zeros into your ring buffer and complete the whole convolution